zjit/src/codegen.rs

```diff
@@ -397,6 +405,7 @@ fn gen_insn(cb: &mut CodeBlock, jit: &mut JITState, asm: &mut Assembler, functio
         &Insn::GuardBlockParamProxy { level, state } => no_output!(gen_guard_block_param_proxy(jit, asm, level, &function.frame_state(state))),
         Insn::PatchPoint { invariant, state } => no_output!(gen_patch_point(jit, asm, invariant, &function.frame_state(*state))),
         Insn::CCall { cfun, args, name: _, return_type: _, elidable: _ } => gen_ccall(asm, *cfun, opnds!(args)),
+        Insn::CCallVariadic { state, .. } if unsafe { rb_c_method_tracing_currently_enabled(jit.ec) } => return Err(*state),
```

Comment:
This seems like a *ton* of plumbing and code changes for what seems to me like it should be a quick change in the optimizer:

* Check if method tracing is enabled. If so, don't optimize to `CCallVariadic`. Add a TODO to add method tracing opcodes later.
* Add a PatchPoint that method tracing is not enabled (if this does not already exist).

Instead of plumbing `ec` through the entire JIT, let's fetch it using `GET_EC` in a ZJIT helper function or something.

Reply:
> to add method tracing opcodes later

Is it for generate optimized code with tracing support? I thought we don't want to compile anything after TracePoint has been enabled?